### PR TITLE
1st half of february cleanup

### PR DESCRIPTION
**Pull request by Derisis13**
Part of my regular QA commits. I attempt to improve code readability and maybe consistency and code size.

### CODE REVIEW

**Comment by Derisis13** [comment]
Should I update `CONTRIBUTING.md` to include a more detailed formatting guideline or should I open another PR?

**Comment by 71zenith** [comment]
> Should I update `CONTRIBUTING.md` to include a more detailed formatting guideline or should I open another PR?

I would recommend making another pr or just adding to #486

**Comment by Derisis13** [comment]
So what it doesn't display here because I made changes locally before pulling iamchokerman's commits is that there was a commit to remove the literal newline in `search_for_unwatched`

File: ani-cli [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
+VERSION='1.6.5'
 
 
-# history file path
-logfile="${XDG_CACHE_HOME:-$HOME/.cache}/ani-hsts"
-auto_play=0
-
 #######################
 # Auxiliary functions #
 #######################
```
Comment (ura43) [inline ani-cli:9]: Can we replace such headings with `### TITLE ###` or `### Title ###` instead?

Comment (Derisis13) [reply ani-cli:9]: Of course!

Comment (Derisis13) [reply ani-cli]: Done in 95ebfa2

File: ani-cli [hunk 2]
```diff
@@ -1,25 +1,37 @@
...
-
 #######################
-# Auxiliary functions #
+# AUXILIARY FUNCTIONS #
 #######################
 
 help_text () {
 	while IFS= read -r line; do
-		printf "%s\n" "$line"
+		echo "$line"
```
Comment (pystardust) [inline ani-cli:29]: why are you using `echo` here? 
Read this https://mywiki.wooledge.org/BashPitfalls#echo_.24foo
Avoid using echo cause the posix implementation of it is questionable. many shells have different implementations of echo even tho they are posix complaint.

File: ani-cli [hunk 3]
```diff
@@ -117,103 +125,85 @@ search_eps () {
...
-	episode_selection
+	process_search
 }
 
 search_history () {
-	tput clear
-	search_results=$(sed -n -E 's/\t[0-9]*//p' "$logfile")
-	[ -z "$search_results" ] && die "History is empty"
-	search_results=$(search_for_unwatched "$search_results")
+	search_results="$(sed -n -E 's/\t[0-9]*//p' "$logfile")"
```
Comment (pystardust) [inline ani-cli:150]: why are you adding quotes here?
Shell word splitting is not applicable here. Word splitting does not occur in variable assignment.
source: posix sh specs 2.9.1 point 2

File: ani-cli [hunk 4]
```diff
@@ -465,44 +435,44 @@ open_episode () {
...
-# to clear the colors when exited using SIGINT
-trap 'printf "\033[0m"; exit 1' INT HUP
+# default options
+player_fn='mpv' #video player needs to be able to play urls
+is_download="0"
+half_ep="0"
+quality='best'
+scrape='query'
+download_dir='.'
+choice=''
```
Comment (pystardust) [inline ani-cli:469]: I see that you have done a lot of
```
variable=''
```
The cleaner way would be
```
variable=
```
just leave a blank

File: ani-cli [hunk 5]
```diff
@@ -465,44 +435,44 @@ open_episode () {
...
-
-# default options
-player_fn="mpv" #video player needs to be able to play urls
-is_download=0
-half_ep=0
-quality=best
-scrape=query
-download_dir="."
-choice=""
+touch "$logfile"
```
Comment (pystardust) [inline ani-cli:477]: why use touch?
It is an external utility. The script was using the beautiful inbuilts of shell scripting.

File: ani-cli [hunk 6]
```diff
@@ -511,27 +481,24 @@ while getopts 'viq:dp:chDUVa:' OPT; do
 			exit 0
 			;;
 		d)
-			is_download=1
+			is_download='1'
```
Comment (pystardust) [inline ani-cli:486]: don't know numbers, we will later use these stuff in `-eq` statements in test. Its better you leave it as is, makes it evident that the variable is a number

**Review by pystardust — CHANGES_REQUESTED** [review]
Try reading the up shell syntax(posix specifications). Quoting everything is definitely safe and nothing is wrong in it but you shouldn't do it when you don't need to.
There is really no reason to quote when there is not field splitting or we certainly know that the variable has a number

**Comment by pystardust** [comment]
I wouldn't suggest this merge, maybe we could discuss here are open a new PR

File: ani-cli (hunk 3 above)
Comment (Derisis13) [reply ani-cli:150]: It was more of a why not then why for me, but I can remove it if you wish

File: ani-cli (hunk 2 above)
Comment (pystardust) [reply ani-cli:29]: basically
```
echo "hi\nthere"
```
word differently in dash and bash

File: ani-cli (hunk 4 above)
Comment (Derisis13) [reply ani-cli:469]: Yes, it does appear cleaner, I'll change it

File: ani-cli (hunk 5 above)
Comment (Derisis13) [reply ani-cli:477]: Shellcheck said not a thing, so I kept it, not knowing it's an external tool. I'll revert it.

File: ani-cli (hunk 6 above)
Comment (Derisis13) [reply ani-cli:486]: You are absolutely right, I'll revert them

File: ani-cli (hunk 3 above)
Comment (pystardust) [reply ani-cli:150]: Check out https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/, this is the specifications of shell scripting. Shell is beautiful, it has a grammar and then syntax that dictates what happens when.

File: ani-cli (hunk 2 above)
Comment (Derisis13) [reply ani-cli]: I thought it was more readable, I wasn't aware how skimpy it is.

**Comment by Derisis13** [comment]
> I wouldn't suggest this merge, maybe we could discuss here are open a new PR

I have implemented your suggestions, but I don't understand why a new PR? Did I change too much?

**Comment by Derisis13** [comment]
Since a formatting guideline was decided not to be included in this PR, I'll mark it as ready in order not to create a bigger mess

**Comment by Derisis13** [comment]
Testing has revealed significant issues, going forward with this PR would be more trouble than what it's worth. Closing PR.

**Pull request closed without merging**